File: sprint3/webRobot/src/main/java/unibo/webRobot/UpdateGui.java
package unibo.webRobot;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.GetMapping;

import java.io.Serializable;

@Controller
public class UpdateGui implements Serializable {
    private String statett;
    private String stateled;
    private String position;
    private String pb;
    private String gb;

    public UpdateGui(){
        this.statett="";
        this.stateled="";
        this.position="";
        this.pb="";
        this.gb="";
    }

    public UpdateGui(String statett,String stateled,String position,String pb,String gb) {
        this.statett=statett;
        this.stateled=stateled;
        this.position=position;
        this.pb=pb;
        this.gb=gb;
    }

    public String getStatett() {
        return statett;
    }

    public String getStateled() {
        return stateled;
    }

    public String getPosition() {
        return position;
    }

    public String getPb() {
        return pb;
    }

    public String getGb() {
        return gb;
    }

    @Override
    public String toString(){
        return "{\"statett\":\""+statett+"\",\"stateled\":\""+stateled+"\",\"position\":\""+position+"\",\"pb\":\""+pb+"\",\"gb\":\""+gb+"\"}";
    }
}

File: sprint3/WebRobotKt/src/main/kotlin/unibo/WebRobotKt/UpdateGui.kt
package unibo.WebRobotKt

import org.springframework.stereotype.Controller
import java.io.Serializable

@Controller
class UpdateGui : Serializable {
    var statett: String
        set(value){
            field=value
        }
    var stateled: String
        set(value){
            field=value
        }
    var position: String
        set(value){
            field=value
        }
    var pb: String
        set(value){
            field=value
        }
    var gb: String
        set(value){
            field=value
        }

    constructor() {
        statett = "coap connecting"
        stateled = "coap connecting"
        position = "coap connecting"
        pb = "coap connecting"
        gb = "coap connecting"
    }

    /*constructor() {
        statett = "load"
        stateled = "OFF"
        position = "HOME"
        pb = "10"
        gb = "10"
    }*/

    override fun toString(): String {
        return "{\"statett\":\"$statett\",\"stateled\":\"$stateled\",\"position\":\"$position\",\"pb\":\"$pb\",\"gb\":\"$gb\"}"
    }
}